### ISSUE TITLE
Submit-ACMEChallenge returns Unexpected Error

### ISSUE THREAD

**Issue by jlcastillo** [ebekker/ACMESharp#64]
I'm following this tutorial:

https://cultiv.nl/blog/lets-encrypt-on-windows/

but when I get to the Submit-ACMEChallenge statement, this is what I get:

Submit-ACMECertificate : Unexpected error
+ $certificateInfo = Submit-ACMECertificate -Ref cert1
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Submit-ACMECertificate], AcmeWebException
    + FullyQualifiedErrorId : ACMESharp.AcmeClient+AcmeWebException,ACMESharp.POSH.SubmitCertificate

How can I fix it?

**Comment by ebekker** [comment]
The unexpected error in this step usually means an unexpected error response from the ACME server (Let's Encrypt).

Are you using the staging or beta server endpoint?

Can you check your system requirements against [these](https://github.com/ebekker/ACMESharp/wiki/Requirements) that I just recently put up?

**Comment by ebekker** [comment]
Also, how many times did you try?  A few users have pointed out that if they get a similar error and then later retry again (either just this step, or the complete series of steps starting with validating a DNS Identifier and all that follow), they get success usually because LE's server was either temporarily down for maintenance or unexpectedly down for issues).

**Comment by jlcastillo** [comment]
I'm using the staging server:

   Initialize-ACMEVault -BaseURI https://acme-staging.api.letsencrypt.org/directory

I tried yesterday and today, with the same result.
I have .NET Framework 4.5.2 and PowerShell 4.0.

**Comment by jlcastillo** [comment]
By the way, I'm using the binaries that I obtained from here:

https://www.kingbain.com/letsencrypt-powershell-modules-for-iis-and-apache/

**Comment by ebekker** [comment]
That version seems fairly recent, there is slightly newer [release](https://github.com/ebekker/ACMESharp/releases) but I suspect it doesn't have much that differs from the release you have.

There have also been more recent changes added to the master branch but again I don't think they will have an impact on your situation.

Can you verify that your site can be externally accessed on port 80 for any file rooted at the following path:  `.well-known/acme-challenge/`

Also, these files should return with a `Content-type` of `text/json`

**Comment by jlcastillo** [comment]
Thanks, I fixed it following your instructions. This is what I did:

1) save the challenge with the extension .json instead of .html. In my case Apache 2.4 sets the right headers automatically.

2) make sure the challenge is properly saved and served. In my case I always store the challenge with the same file name, and I proxy all requests that look like ".well-known/acme-challenge" to this file. But this challenge file was not overwritten with every execution, therefore the challenge that I served was old, from a previous execution.
